api/src/main/java/org/lfenergy/shapeshifter/api/model/UftpParticipantInformation.java

```diff
@@ -4,4 +4,4 @@
 
 package org.lfenergy.shapeshifter.api.model;
 
-public record UftpParticipantInformation(String domain, String publicKey, String endpoint) {}
+public record UftpParticipantInformation(String domain, String publicKey, String endpoint, boolean requiresAuthorization) {}
```

Comment:
Maybe add some Javadoc on this class and the parameters to indicate how each value is used. Specifically what happens if you set requiresAuthorization to true.